src/environmentd/tests/sql.rs

```diff
@@ -2430,3 +2430,71 @@ fn test_query_on_dropped_source() {
         )
     });
 }
+
+#[test]
+fn test_dont_drop_sinks_twice() {
+    let config = util::Config::default().workers(4);
+    let server = util::start_server(config).unwrap();
+
+    let (notice_tx, mut notice_rx) = futures::channel::mpsc::unbounded();
+
+    let (create_tx, create_rx) = futures::channel::oneshot::channel();
+    let (drop_tx, drop_rx) = futures::channel::oneshot::channel();
+
+    let mut client_a = server
+        .pg_config()
+        .notice_callback(move |notice| {
+            notice_tx.unbounded_send(notice).unwrap();
+        })
+        .connect(postgres::NoTls)
+        .unwrap();
+
+    let handle_a = std::thread::spawn(move || {
+        client_a.batch_execute("CREATE TABLE t1 (a INT)").unwrap();
+        client_a.batch_execute("CREATE TABLE t2 (a INT)").unwrap();
+
+        create_tx.send(()).unwrap();
+
+        let _out = client_a
+            .copy_out("COPY (SUBSCRIBE (SELECT * FROM t1,t2)) TO STDOUT")
+            .unwrap();
+
+        // Keep the subscribe alive until we drop the underlying tables.
+        futures::executor::block_on(drop_rx).unwrap();
+
+        // Drop our client so the notice channel closes.
+        drop(_out);
+        drop(client_a);
```

Comment:
Is there any difference between explicitly dropping here or just letting them fall out of scope?

Reply:
Good catch, at the time there wasn't, not sure why I had this haha